sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/implementation/VertxHttpResponseBase.java

```diff
@@ -56,6 +53,18 @@ public final Mono<String> getBodyAsString() {
 
     @Override
     public final Mono<String> getBodyAsString(Charset charset) {
-        return getBodyAsByteArray().map(bytes -> CoreUtils.bomAwareToString(bytes, charset.toString()));
+        return getBodyAsByteArray().map(bytes -> new String(bytes, charset));
+    }
+
+    /**
+     * Creates azure-core HttpHeaders from Vert.x HTTP headers.
+     *
+     * @param vertxHttpHeaders Vert.x HTTP response headers.
+     * @return Azure HTTP headers.
+     */
+    private HttpHeaders fromVertxHttpHeaders(MultiMap vertxHttpHeaders) {
+        HttpHeaders azureHttpHeaders = new HttpHeaders();
+        vertxHttpHeaders.names().forEach(name -> azureHttpHeaders.set(name, vertxHttpHeaders.getAll(name)));
+        return azureHttpHeaders;
```

Comment:
This implementation may be better performing:

```java
HttpHeaders azureHeaders = new HttpHeaders(headers.size());
headers.forEach(azureHeaders::add);
return azureHeaders;
```

The current implementation needs to parse all headers once to build a Set of names then iterate over them each time getting all values for the header name. Our most common case is each header is only see once, and even if it isn't this should still be much cheaper. The only case where this will have worse performance is if there is a lot of repeated header names.